### PR TITLE
add change migrations and corresponding diesel struct

### PR DESCRIPTION
**Pull request by khodzha**

### CODE REVIEW

File: src/db/agent.rs [hunk 1]
```diff
@@ -11,11 +11,10 @@ use crate::schema::agent;
 ////////////////////////////////////////////////////////////////////////////////
 
 #[derive(Clone, Copy, Debug, DbEnum, Deserialize, Serialize, PartialEq)]
-#[serde(rename_all = "lowercase")]
+#[serde(rename_all = "snake_case")]
 #[PgType = "agent_status"]
 #[DieselType = "Agent_status"]
 pub(crate) enum Status {
-    #[serde(rename = "in_progress")]
```
Comment (khodzha) [inline src/db/agent.rs]: removed this cause this cased is handled by `rename_all = "snake_case"`

File: migrations/2020-03-11-090819_create_changes/up.sql [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    event_id uuid,
+    created_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
+
+    FOREIGN KEY (edition_id) REFERENCES edition (id) ON DELETE CASCADE,
+    FOREIGN KEY (event_id) REFERENCES event (id) ON DELETE CASCADE,
+    PRIMARY KEY (id)
+);
+
+CREATE INDEX change_edition_id_idx ON change (edition_id);
+CREATE INDEX change_event_id_idx ON change (event_id);
```
Comment (feymartynov) [inline migrations/2020-03-11-090819_create_changes/up.sql]: Why do we need this index?

File: migrations/2020-03-11-090819_create_changes/up.sql [hunk 3]
```diff
@@ -0,0 +1,17 @@
+CREATE TYPE change_type AS ENUM ('addition', 'modification', 'removal');
+
+CREATE TABLE change (
+    id uuid default gen_random_uuid(),
+    edition_id uuid NOT NULL,
+    kind change_type NOT NULL,
+    event jsonb NOT NULL DEFAULT '{}'::jsonb,
```
Comment (feymartynov) [inline migrations/2020-03-11-090819_create_changes/up.sql]: I think this is better to call `data` to keep in sync with `event.data` field.

File: migrations/2020-03-11-090819_create_changes/up.sql [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    event_id uuid,
```
Comment (feymartynov) [inline migrations/2020-03-11-090819_create_changes/up.sql]: Isn't it optional in case of addition?

Comment (khodzha) [reply migrations/2020-03-11-090819_create_changes/up.sql]: how does one express optionality? it isnt `not null` right now :thinking:

File: migrations/2020-03-11-090819_create_changes/up.sql (hunk 3 above)
Comment (khodzha) [reply migrations/2020-03-11-090819_create_changes/up.sql]: agreed

File: migrations/2020-03-11-090819_create_changes/up.sql (hunk 2 above)
Comment (khodzha) [reply migrations/2020-03-11-090819_create_changes/up.sql]: agreed, probably not needed

**Comment by feymartynov** [comment]
Don't forget to run `cargo fmt` before merging.

**Pull request merged**